lib/Haptics.tsx

```diff
@@ -0,0 +1,78 @@
+import * as ExpoHaptics from "expo-haptics"
+import { ReactNode, createContext, useContext } from "react"
+import { HapticsManager } from "./HapticsManager"
+
+/**
+ * A haptic event to be played to the user.
+ */
+export interface HapticsFunctions {
+  play: (hapticTrigger: HapticEvent) => Promise<void>
+  mute: () => void
+  unmute: () => void
+}
+
+/**
+ * A function type to indicate haptic feedback being played to the user.
+ */
+export type HapticEvent =
+  | "light tap"
+  | "medium tap"
+  | "heavy tap"
+  | "selection"
+  | "success"
+  | "warning"
+  | "error"
+
+const HapticsManagerContext = createContext<HapticsManager | undefined>(
+  undefined
+)
+
+export const useHapticsFunctions = () => useContext(HapticsManagerContext)!
+
+export type HapticsProviderProps = HapticsManager & {
+  children: ReactNode
+}
```

Comment:
Instead of adding in HapticsManager, add a haptics property with the interface used here